functions/uploads.py

```diff
@@ -55,19 +81,22 @@ def ingest_upload(event: dict, context: BackgroundContext):
             "Merging metadata from upload %d into trial %s: " % (job.id, trial_id),
             metadata_with_urls,
         )
-        TrialMetadata.patch_assays(trial_id, metadata_with_urls, session=session)
+        with saved_failure_status(job, session):
+            TrialMetadata.patch_assays(trial_id, metadata_with_urls, session=session)
 
         # Save downloadable files to the database
         # NOTE: this needs to happen after TrialMetadata.patch_assays
         # in order to avoid violating a foreign-key constraint on the trial_id
         # in the event that this is the first upload for a trial.
         for artifact_metadata in downloadable_files:
-            print(
-                f"Saving metadata for {target_url} to downloadable_files table: {artifact_metadata}"
-            )
-            DownloadableFiles.create_from_metadata(
-                trial_id, job.assay_type, artifact_metadata, session=session
-            )
+            print(f"Saving metadata to downloadable_files table: {artifact_metadata}")
+            with saved_failure_status(job, session):
+                DownloadableFiles.create_from_metadata(
+                    trial_id, job.assay_type, artifact_metadata, session=session
+                )
+
+    # Save the upload success
+    job.status = AssayUploadStatus.MERGE_COMPLETED.value
```

Comment:
Seems this lies out of `with sqlalchemy_session() as session:` so I'm not sure if this change will be committed